knimin/templates/pm_plate_map.html

```diff
@@ -0,0 +1,358 @@
+    background-color: white;
+  }
+  input[type="text"] {
+    border-radius: 0;
+  }
+  .chosen-container-single .chosen-single {
+    /* height: 30px;
+    border-radius: 3px; */
+    height: 25px;
+    border-radius: 0;
+    background-image: none;
+    border: 1px solid black;
+    margin-bottom: 1px;
+    font-weight: normal;
+    font-size: 90%;
+  }
+  .chosen-container-single .chosen-results {
+    font-weight: normal;
+    font-size: 90%;
+  }
+  .chosen-choices {
+      border: 1px solid #ccc;
+      border-radius: 4px;
+      min-height: 34px;
+      padding: 6px 12px;
+  }
+</style>
+<script type="text/javascript">
+
+  // Retrieve data
+  var target = {{dumps(target)}};                 // Target object: sample, primer, barseq, etc.
+  var id = {{dumps(id)}};                         // Plate ID (0 for creating new plate)
+  var info = {{dumps(info)}};                     // Plate info: name, email, created_on, notes
+  var type = {{dumps(type)}};                     // Plate type: name, cols, rows, notes
+  var layout = {{dumps(layout)}};                 // Plate layout: sample_id, col, row, name, notes
+  var currentuser = {{dumps(currentuser)}};       // Current user
+
+  // Pre-process data
+  // The coordinate of a well is represented by a scalar "col_row" to facilitate searching
+  var well2id = {};
+  for (var i = 0; i < layout.length; i++) {
+    well2id[layout[i]['col'].toString() + "_" + layout[i]['row'].toString()] = i;
+  }
+
+  $(document).ready(function() {
+
+    // Render display items
+    $(".chosen-select").chosen();
+    $("#plated_on").datepicker();
+    $.ajax({
+      url: "/pm_plate_map/",
+      data: {action: "emails"},
+      success: function(emails) {
+        emails = JSON.parse(emails)
+        for (var i = 0; i < emails.length; i++) {
+          $("#plated_by").append($("<option>", {value: emails[i], text: emails[i]}));
+        }
+        $("#plated_by").trigger("chosen:updated");
+      }
+    });
+    $.ajax({
+      url: "/pm_plate_map/",
+      data: {action: "plate_types"},
+      success: function(plate_types) {
+        plate_types = JSON.parse(plate_types)
+        for (var i = 0; i < plate_types.length; i++) {
+          $("#plate_types").append($("<option>", {value: plate_types[i], text: plate_types[i]}));
+        }
+        $("#plate_types").val(type["name"]);
+        $("#plate_types").trigger("chosen:updated");
+      }
+    });
+
+    if (id > 0) {  // View or edit an existing plate
+      $("#plate_id").html("Plate # " + id.toString() + ": ");
+      $("#plate_name").val(info["name"]);
+      $("#plated_by").val(info["email"]);
+      $("#plated_on").datepicker("setDate", info["created_on"]);
+      $("#spanSave").hide();
+    } else {  // Create a new plate
+      $("#plated_by").val(currentuser);
+      $("#plated_on").datepicker("setDate", new Date());
+      $("#spanNav").hide();
+    }
+
+    // Generate plate map
+    var table = document.getElementById("plate_map");
```

Comment:
Ok, I think I understand what you are doing and that's why you needed to use the dumps/loads in here. You are actually using JavaScript to generate the page, instead of using tornado's template system. I think it is better to use tornado's templating system rather than this route. What do others think?